### PR TITLE
Tunic sponsor modal

### PR DESCRIPTION
**Pull request by mparent61**
Rough, needs some @kand TLC (see comments)

Additionally, this approach duplicates logic from Special Coverage implementation, ideally should share this logic.

### CODE REVIEW

File: app/scripts/controllers/sponsormodal.js [hunk 1]
```diff
@@ -1,20 +1,63 @@
...
+                                            // NOTE: temp depdendencies will be moved along with searchCampaigns/loadCampaigns
+                                            _, $q, $http, TunicConfig) {
     $scope.article = article;
 
-    if ($scope.article.campaign) {
-      $scope.campaign = Campaign.$find($scope.article.campaign);
+    $scope.tunicCampaignIdMapping = {};
+
+    // --------------------------- TODO: MOVE ME SOMEWHERE ELSE---------------------------------------
+    var $loadTunicCampaign = function (tunicCampaignId) {
```
Comment (mparent61) [inline app/scripts/controllers/sponsormodal.js]: Where to put this (and $searchCampaigns)? Special Coverage implementation added this to model inside special-coverage-factory.js). Or better to share this logic somehow?

File: app/scripts/controllers/sponsormodal.js [hunk 2]
```diff
@@ -1,20 +1,63 @@
...
+        params: params,
+      }).then(function (response) {
+        return response.data.results;
+      });
+    };
+    // --------------------------- END TODO ---------------------------------------
+
+    if ($scope.article.tunic_campaign_id) {
+      $loadTunicCampaign($scope.article.tunic_campaign_id).then(function (campaign) {
+        $scope.tunicCampaignId = campaign.id;
```
Comment (mparent61) [inline app/scripts/controllers/sponsormodal.js]: Do I even need to set this, or can i just reference $scope.article.tunic_campaign_id inside the HTML element?

File: app/scripts/controllers/sponsormodal.js [hunk 3]
```diff
@@ -1,20 +1,63 @@
...
+    var $searchCampaigns = function (params) {
+      return $http.get(TunicConfig.buildBackendApiUrl('campaign/'), {
+        params: params,
+      }).then(function (response) {
+        return response.data.results;
+      });
+    };
+    // --------------------------- END TODO ---------------------------------------
+
+    if ($scope.article.tunic_campaign_id) {
```
Comment (mparent61) [inline app/scripts/controllers/sponsormodal.js]: For some reason this member is not camel-cased...

**Comment by kand** [comment]
Closing in favor of https://github.com/theonion/bulbs-cms/pull/118

**Pull request closed without merging**